### PR TITLE
Add softhsm

### PR DESCRIPTION
**Pull request by xen0l**

### CODE REVIEW

File: components/softhsm/Makefile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH=	\
+    sha256:eae8065f6c472af24f4c056d6728edda0fd34306f41a818697f765a6a662338d
+COMPONENT_ARCHIVE_URL= https://dist.opendnssec.org/source/$(COMPONENT_ARCHIVE)
+
+include $(WS_TOP)/make-rules/prep.mk
+include $(WS_TOP)/make-rules/configure.mk
+include $(WS_TOP)/make-rules/ips.mk
+
+COMPONENT_BUILD_ARGS+= -j8
```
Comment (xen0l) [inline components/softhsm/Makefile]: This will be removed

File: components/softhsm/softhsm.p5m [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+set name=pkg.fmri value=pkg:/security/softhsm@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value=org.opensolaris.category.2008:System/Security
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license LICENSE license='$(COMPONENT_LICENSE)'
+
+dir  path=etc
```
Comment (pyhalov) [inline components/softhsm/softhsm.p5m]: we usually avoid shipping dir actions

File: components/softhsm/softhsm.p5m [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+file path=etc/softhsm2.conf
```
Comment (pyhalov) [inline components/softhsm/softhsm.p5m]: preserve=true?

File: components/softhsm/softhsm.p5m [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+dir  path=usr
+dir  path=usr/bin
+file path=usr/bin/softhsm2-dump-file
+file path=usr/bin/softhsm2-keyconv
+file path=usr/bin/softhsm2-util
+dir  path=usr/lib
+dir  path=usr/lib/softhsm
+file path=usr/lib/softhsm/libsofthsm2.a
```
Comment (pyhalov) [inline components/softhsm/softhsm.p5m]: Do we need .a library? I don't see sense in delivering it if we deliver corresponding dll

File: components/softhsm/softhsm.p5m [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+dir  path=usr/share/man/man1
+file path=usr/share/man/man1/softhsm2-dump-file.1
+file path=usr/share/man/man1/softhsm2-keyconv.1
+file path=usr/share/man/man1/softhsm2-util.1
+dir  path=usr/share/man/man5
+file path=usr/share/man/man5/softhsm2.conf.5
+dir  path=var
+dir  path=var/lib
+dir  path=var/lib/softhsm
+dir  path=var/lib/softhsm/tokens
```
Comment (pyhalov) [inline components/softhsm/softhsm.p5m]: What is this directory used for? Seems to be a bit Linux-like to deliver something in /var/lib.

File: components/softhsm/Makefile [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+CONFIGURE_OPTIONS+= --sysconfdir=/etc
+CONFIGURE_OPTIONS+= --localstatedir=/var
+CONFIGURE_OPTIONS+= --disable-ecc
+CONFIGURE_OPTIONS+= --disable-gost
+CONFIGURE_OPTIONS+= --with-crypto-backend=botan
+
+build:		$(BUILD_32)
```
Comment (pyhalov) [inline components/softhsm/Makefile]: Don't we need also 64-bit version?

File: components/softhsm/Makefile [hunk 7]
```diff
@@ -0,0 +1,43 @@
+#
```
Comment (pyhalov) [inline components/softhsm/Makefile]: please, move it to corresponding subdirectory

**Pull request merged**